panda/src/cocoadisplay/cocoaGraphicsStateGuardian.h

```diff
@@ -45,6 +47,13 @@ class CocoaGraphicsStateGuardian : public GLGraphicsStateGuardian {
   NSOpenGLContext *_share_context;
   NSOpenGLContext *_context;
   FrameBufferProperties _fbprops;
+  
+  CVDisplayLinkRef _display_link;
+  Mutex swap_lock;
+  ConditionVar swap_condition;
+  std::atomic_bool should_wait;
```

Comment:
You should use an `std::atomic_flag` for this.  The only operations you are using are `clear()` and `test_and_set()`.